sklearn/neighbors/base.py

```diff
@@ -323,7 +321,7 @@ class from an array representing our data set and ask who's
                [2]]...)
 
         """
-        if self._fit_method is None:
+        if check_is_fitted(self, "_fit_method"):
```

Comment:
`check_is_fitted(self, "_fit_method")` already raises an error, the if + raise is not useful.